lib/processor.js

```diff
@@ -36,30 +36,53 @@ exports.work = (audioEvent) => {
   });
 }
 
-// upload to final s3 destination
-exports.upload = (localPath, filePath, fileName) => {
-  let localStream = fs.createReadStream(localPath);
-  let fileParts = path.parse(fileName);
+// upload mp3 and make copies as needed
+exports.uploadAndCopy = (file) => {
+  let dest = `${file.path}/${file.name}`;
+  let type = file.contentType;
+  if (!type && file.format) {
+     type = `audio/${file.format}`;
+  }
+  let uploadOriginal;
+  if (file.s3Bucket && file.s3Key) {
+    uploadOriginal = this.copy(file.s3Bucket, file.s3Key, dest, type);
+  } else {
+    uploadOriginal = this.upload(file.localPath, dest, type);
+  }
+
+  let fileParts = path.parse(file.name);
   let fileCopies = COPY_VARIATIONS.map(vary => {
     return `${fileParts.name}_${vary}${fileParts.ext}`;
   });
 
-  // first upload the original file
+  return uploadOriginal.then(data => {
+    let makeCopies = fileCopies.map(name => {
+      this.copy(data.Bucket, data.Key, `${file.path}/${name}`, type);
+    });
+    return Q.all(makeCopies).then(() => [file.name].concat(fileCopies));
+  });
+};
+
+// upload from local to s3
+exports.upload = (localPath, destKey, contentType) => {
   return Q.ninvoke(s3, 'upload', {
-    Body: localStream,
+    Body: fs.createReadStream(localPath),
     Bucket: process.env.DESTINATION_BUCKET,
-    Key: `${filePath}/${fileName}`
-  }).then(data => {
-    let makeCopies = fileCopies.map(copyName => {
-      return Q.ninvoke(s3, 'copyObject', {
-        CopySource: `${data.Bucket}/${data.Key}`,
-        Bucket: process.env.DESTINATION_BUCKET,
-        Key: `${filePath}/${copyName}`
-      });
-    });
-    return Q.all(makeCopies).then(() => {
-      return [fileName].concat(fileCopies);
-    });
+    Key: destKey,
+    ContentType: contentType
+  })
+}
+
+// copy between s3 locations
+exports.copy = (fromBucket, fromKey, destKey, contentType) => {
+  return Q.ninvoke(s3, 'copyObject', {
+    CopySource: `${fromBucket}/${fromKey}`,
+    Bucket: process.env.DESTINATION_BUCKET,
+    Key: destKey,
+    MetadataDirective: 'REPLACE',
```

Comment:
When doing a `copyObject`, it actually ignores the `ContentType` set here unless you use REPLACE.  I think this is fine though, since we don't really want the source's S3 metadata anyways.